src/pages/MarketDetails/useGetMarketData.ts

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import BigNumber from 'bignumber.js';
+
+import { IVBepToken } from 'types';
+import { getToken } from 'utilities';
+import { convertWeiToCoins } from 'utilities/common';
+import { VTOKEN_DECIMALS } from 'config';
+import { useGetMarkets } from 'clients/api';
+
+const useGetMarketData = ({
+  vTokenId,
+  vTokenAddress,
+}: {
+  vTokenId: IVBepToken['id'];
+  vTokenAddress: IVBepToken['address'];
+}) => {
+  const { data: getMarketData } = useGetMarkets();
```

Comment:
I would put this in the hooks directory. Currently we merge markets and assets in the `useUserMarketHook`.

I think it would make sense to include logic for setting units in the `useGetMarkets` hook so we sanitize and format data the way we want to use it before it is ever used. 

How we handle / merge assets and markets could probably be further optimized and ties into an on going discussion with Lakshmi

Reply:
This hook will only be relevant to this component, so I don't think it should be put in the `hooks` directory (since it won't be reused).

About setting the units in the `useGetMarkets` hook, for now I wouldn't spend time on it since this is bound to be replaced once we have the subgraph in place and we move away from using the backend at all.

Yeah I agree. For now I think we've been following the right approach and I'd continue that way: we use React Query to refactor the existing fetching functions, so that we benefit from caching and all the rest, but don't actually modify the data model until we are done with the UI refresh.

Reply:
We have been making minor changes to the data model for example with units and adding to the userMarket hook.

I'm fine with leaving this as is for now but I think it will get refactored as you said and we should look at how useUserMarkets is accessing this same information.